src/Data/Aeson/TH.hs

```diff
@@ -1150,27 +1139,30 @@ parseTypeMismatch tName conName expected actual =
           ]
 
 class LookupField a where
-    lookupField :: (Value -> Parser a) -> String -> String
+    lookupField :: Maybe a -> (Value -> Parser a) -> String -> String
```

Comment:
This whole class and all of it's instances seem a bit fishy to me. I think it's saying "Use `lookupFieldWith` if you don't know about the type, but if you *know* it's a `Maybe` or `Option`, then do `parseOptionalFieldWith`."

This suggests another approach: define `instance LookupField NullableNonEmptyText` that is optional. `LookupField` is (sensibly) not exported, so we'd have to export it, and the class/instance design makes that *pretty* unsafe. Still, it is an alternative.

Reply:
(Notably, this *does not work* for `Generic`)